### PR TITLE
ddl: support create sequences with `BatchCreateTableWithInfo`

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/53374

Problem Summary:

### What changed and how does it work?

- Invoke `callHookOnChanged` after `BatchCreateTableWithInfo`.
- Refactor and extract `createSequenceWithCheck`.
- Support create sequences in `onCreateTables`.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/sequence.go [hunk 1]
```diff
@@ -45,41 +45,40 @@ func onCreateSequence(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ err
...
+	case model.StateNone, model.StatePublic:
 		// none -> public
 		tbInfo.State = model.StatePublic
 		tbInfo.UpdateTS = t.StartTS
-		err = createSequenceWithCheck(t, job, schemaID, tbInfo)
+		err := checkTableInfoValid(tbInfo)
 		if err != nil {
-			return ver, errors.Trace(err)
+			job.State = model.JobStateCancelled
+			return errors.Trace(err)
```
Comment (AilinKid) [inline pkg/ddl/sequence.go:70]: LGTM

**Review by AilinKid — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

File: pkg/ddl/sequence.go [hunk 2]
```diff
@@ -45,41 +45,40 @@ func onCreateSequence(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ err
...
 		return ver, errors.Trace(err)
 	}
+	job.FinishTableJob(model.JobStateDone, model.StatePublic, ver, tbInfo)
+	return ver, nil
+}
 
+func createSequenceWithCheck(t *meta.Meta, job *model.Job, tbInfo *model.TableInfo) error {
 	switch tbInfo.State {
-	case model.StateNone:
+	case model.StateNone, model.StatePublic:
```
Comment (wjhuang2016) [inline pkg/ddl/sequence.go:63]: Why add model.StatePublic here?

Comment (tangenta) [reply pkg/ddl/sequence.go:63]: The reason is the same as https://github.com/pingcap/tidb/pull/51870. Maybe we can set this state on caller side, let's refactor these in next PR.

**Review by wjhuang2016 — APPROVED** [review]

**Comment by tangenta** [comment]
Encounter an unstable test that will be fixed by #53566

**Pull request merged**